### PR TITLE
Feature: Custom hotKeys (manager only)

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
Prepare custom hotkeys
* Add Code and Label to HotKey. Label is localized, but Code is language independent.
* Add IsVisible to HotKey. It is possible to define the visibility for each hotkey. If not visible, the code starts with '!'.
* Add HotKey.Parse(code).
* Moved code for HotKey.Label, for more structured code.
* Add Mode=OneWay to each use of HotKeyText and LabelWithHotKey.
* Add HotKeyCollection(immutable) with Code, Label and Parse(Code). To parse, hotkeys can be separated by ',', ';', spaces or tab.
* IRichCommand no longer exposes hotkeys one by one, but HotKeys (HotKeyCollection).
* DefaultHotKeys are no longer exposed by commands. This is to avoid confusion. Non-custom commands can be modified to release a hotkey used by another command.
* Edit HotKeys removes this hotkeys from its old command. A hotkey can only have one command.
* It is not yet possible to modify hotkeys from the application.
* Saved in user_settings.json. It is possible to manually edit the file to modify the hotkeys. You need to know the syntax.

Later
* Add a page in the settings to easily modify custom hotkeys. It will be necessary to warn if the hotkey is already used.
* Check that each button or menu that uses a command displays its hotkeys, even if the command has none by default.

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
To test, we must add code to modify CustomHotKeys (using HotKeyCollection.Parse).

### CODE REVIEW

**Comment by yaira2** [comment]
> Save custom hotkeys to a configuration file.

This should be done in the settings file so that users can modify everything together.

**Comment by cinqmilleans** [comment]
@yair100 It's done. The description is updated. The next step is to create the settings page. I'm taking care of it for a next pr.

File: src/Files.Backend/Services/Settings/IPreferencesSettingsService.cs [hunk 1]
```diff
@@ -176,5 +176,10 @@ public interface IPreferencesSettingsService : IBaseSettingsService, INotifyProp
 		/// A dictionary to determine which hashes should be shown.
 		/// </summary>
 		Dictionary<string, bool> ShowHashesDictionary { get; set; }
+
+		/// <summary>
+		/// A dictionary to determine the custom hotkeys
+		/// </summary>
+		Dictionary<string, string> CustomHotKeys { get; set; }
```
Comment (yaira2) [inline src/Files.Backend/Services/Settings/IPreferencesSettingsService.cs:183]:
```suggestion
		Dictionary<string, string> Actions { get; set; }
```

Comment (cinqmilleans) [reply src/Files.Backend/Services/Settings/IPreferencesSettingsService.cs]: It's done.

File: src/Files.App/Commands/HotKey/HotKey.cs [hunk 2]
```diff
@@ -1,38 +1,301 @@
...
 	{
+		private static readonly IImmutableDictionary<KeyModifiers, string> modifiers = new Dictionary<KeyModifiers, string>()
+		{
+			[KeyModifiers.Menu] = GetKeyString("Menu"),
+			[KeyModifiers.Ctrl] = GetKeyString("Control"),
+			[KeyModifiers.Shift] = GetKeyString("Shift"),
+			[KeyModifiers.Win] = GetKeyString("Windows"),
+		}.ToImmutableDictionary();
+
+		private static readonly IImmutableDictionary<Keys, string> keys = new Dictionary<Keys, string>()
```
Comment (d2dyno1) [inline src/Files.App/Commands/HotKey/HotKey.cs:24]: This doesn't feel right to me. Maybe you could iterate over the enum values and call `.ToString()` on them

Comment (cinqmilleans) [reply src/Files.App/Commands/HotKey/HotKey.cs:24]: These texts are localized because the keys do not have the same name in all languages.

Comment (d2dyno1) [reply src/Files.App/Commands/HotKey/HotKey.cs:24]: ehhh

**Review by d2dyno1 — DISMISSED** [review]
I don't have anything more to add 👍

File: src/Files.App/Commands/Manager/CommandManager.cs [hunk 3]
```diff
@@ -310,75 +353,135 @@ private class NoneCommand : IRichCommand
...
+				customs.Remove(Code.ToString());
+				manager.settings.Actions = customs;
+			}
+
+			internal void UpdateHotKeys(bool isCustom, HotKeyCollection hotKeys)
+			{
+				if (Code is CommandCodes.SelectAll or CommandCodes.ClearSelection)
+				{
+
+				}
```
Comment (QuaintMako) [inline src/Files.App/Commands/Manager/CommandManager.cs:474]: What is the use of that empty condition?

Comment (d2dyno1) [reply src/Files.App/Commands/Manager/CommandManager.cs:474]: Good catch!

Comment (cinqmilleans) [reply src/Files.App/Commands/Manager/CommandManager.cs]: Sorry. It's fixed.

**Comment by cinqmilleans** [comment]
Merge fixed.

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**